v11.11.0 proposal

BridgeAR opened the pull request:
### Notable Changes

* **n-api**:
  * Implement date object (Jarrod Connolly) [#25917](https://github.com/nodejs/node/pull/25917)
* **util**:
  * Add compact depth mode for `util.inspect()` (Ruben Bridgewater) [#26269](https://github.com/nodejs/node/pull/26269)
* **worker**:
  * Improve integration with native addons (Anna Henningsen) [#26175](https://github.com/nodejs/node/pull/26175)
  * MessagePort.prototype.onmessage takes arguments closer to the Web specification now (Anna Henningsen) [#26082](https://github.com/nodejs/node/pull/26082)

### Commits

* [[`d66cb4a116`](https://github.com/nodejs/node/commit/d66cb4a116)] - **benchmark,doc,lib,test**: capitalize comments (Ruben Bridgewater) [#26223](https://github.com/nodejs/node/pull/26223)
* [[`f4955fde60`](https://github.com/nodejs/node/commit/f4955fde60)] - **benchmark,test**: refactoring (Refael Ackermann) [#26119](https://github.com/nodejs/node/pull/26119)
* [[`5e4aa28e1c`](https://github.com/nodejs/node/commit/5e4aa28e1c)] - **buffer**: avoid materializing ArrayBuffer for creation (Anna Henningsen) [#26301](https://github.com/nodejs/node/pull/26301)
* [[`05e6ec0143`](https://github.com/nodejs/node/commit/05e6ec0143)] - **build**: make 'floating patch' message informational (Ben Noordhuis) [#26349](https://github.com/nodejs/node/pull/26349)
* [[`e2baa6836b`](https://github.com/nodejs/node/commit/e2baa6836b)] - **build**: remove v8\_typed\_array\_max\_size\_in\_heap option (Anna Henningsen) [#26301](https://github.com/nodejs/node/pull/26301)
* [[`fa8110a60e`](https://github.com/nodejs/node/commit/fa8110a60e)] - **build**: silence cpp lint by default (Ruben Bridgewater) [#26252](https://github.com/nodejs/node/pull/26252)
* [[`dbbcedae6d`](https://github.com/nodejs/node/commit/dbbcedae6d)] - **build**: tidy up comments in `create\_expfile.sh` (Richard Lau) [#26220](https://github.com/nodejs/node/pull/26220)
* [[`f408d78914`](https://github.com/nodejs/node/commit/f408d78914)] - **build**: fixed clang's warning when building openssl (Thang Tran) [#25954](https://github.com/nodejs/node/pull/25954)
* [[`a3f7471d35`](https://github.com/nodejs/node/commit/a3f7471d35)] - **build,test**: guard eslint with crypto check (Daniel Bevenius) [#26182](https://github.com/nodejs/node/pull/26182)
* [[`a70bafb3cc`](https://github.com/nodejs/node/commit/a70bafb3cc)] - **console**: prevent constructing console methods (Thomas) [#26096](https://github.com/nodejs/node/pull/26096)
* [[`1333dccede`](https://github.com/nodejs/node/commit/1333dccede)] - **crypto**: fix unencrypted DER PKCS8 parsing (Tobias Nießen) [#26236](https://github.com/nodejs/node/pull/26236)
* [[`70e463c294`](https://github.com/nodejs/node/commit/70e463c294)] - **crypto**: fix error condition in Verify::VerifyFinal (Tobias Nießen) [#26238](https://github.com/nodejs/node/pull/26238)
* [[`108c698f44`](https://github.com/nodejs/node/commit/108c698f44)] - **crypto**: make ConvertKey clear openssl error stack (Ben Noordhuis) [#26153](https://github.com/nodejs/node/pull/26153)
* [[`c8d30a7313`](https://github.com/nodejs/node/commit/c8d30a7313)] - **deps**: update acorn to 6.1.0 (gengjiawen) [#26102](https://github.com/nodejs/node/pull/26102)
* [[`7f08e0238a`](https://github.com/nodejs/node/commit/7f08e0238a)] - **deps**: V8: cherry-pick d3308d0 (Anna Henningsen) [#26207](https://github.com/nodejs/node/pull/26207)
* [[`206e4b043b`](https://github.com/nodejs/node/commit/206e4b043b)] - **deps**: V8: backport 74571c8 (Ruben Bridgewater) [#25941](https://github.com/nodejs/node/pull/25941)
* [[`f0a81664c7`](https://github.com/nodejs/node/commit/f0a81664c7)] - **deps**: backport ICU fix for ARM64 Windows (Jon Kunkee) [#26090](https://github.com/nodejs/node/pull/26090)
* [[`ea26ac0f2b`](https://github.com/nodejs/node/commit/ea26ac0f2b)] - **dns**: refactor QueryWrap lifetime management (Anna Henningsen) [#26253](https://github.com/nodejs/node/pull/26253)
* [[`846cba056e`](https://github.com/nodejs/node/commit/846cba056e)] - **doc**: edit deprecation identifier info in Collaborator Guide (Rich Trott) [#26372](https://github.com/nodejs/node/pull/26372)
* [[`3f4b27d681`](https://github.com/nodejs/node/commit/3f4b27d681)] - **doc**: maxReservedRemoteStreams value constraints (Sebastiaan Deckers) [#26309](https://github.com/nodejs/node/pull/26309)
* [[`bc5771ec91`](https://github.com/nodejs/node/commit/bc5771ec91)] - **doc**: correct typos in various docs (Beni von Cheni) [#26312](https://github.com/nodejs/node/pull/26312)
* [[`3560c3abeb`](https://github.com/nodejs/node/commit/3560c3abeb)] - **doc**: sort http.request() options alphabetically (Luigi Pinca) [#26152](https://github.com/nodejs/node/pull/26152)
* [[`86982558ad`](https://github.com/nodejs/node/commit/86982558ad)] - **doc**: add documentation for the defaultPort option (Luigi Pinca) [#26152](https://github.com/nodejs/node/pull/26152)
* [[`7bf6309f0b`](https://github.com/nodejs/node/commit/7bf6309f0b)] - **doc**: napi\_get\_value\_bigint\_words argument order (Michael Wei) [#26300](https://github.com/nodejs/node/pull/26300)
* [[`40a5a93b41`](https://github.com/nodejs/node/commit/40a5a93b41)] - **doc**: add example for setting Vary: Accept-Encoding header in zlib.md (Mukul Khanna) [#26308](https://github.com/nodejs/node/pull/26308)
* [[`85840681a4`](https://github.com/nodejs/node/commit/85840681a4)] - **doc**: revise deprecation semverness info in Collaborator Guide (Rich Trott) [#26232](https://github.com/nodejs/node/pull/26232)
* [[`ff57a1c321`](https://github.com/nodejs/node/commit/ff57a1c321)] - **doc**: clarify http.ClientRequest path description (Minwoo Jung) [#26259](https://github.com/nodejs/node/pull/26259)
* [[`5e44768e9f`](https://github.com/nodejs/node/commit/5e44768e9f)] - **doc**: revise deprecation level explanations in Collaborator Guide (Rich Trott) [#26197](https://github.com/nodejs/node/pull/26197)
* [[`823f0ce952`](https://github.com/nodejs/node/commit/823f0ce952)] - **doc**: revise Style Guide (Rich Trott) [#26176](https://github.com/nodejs/node/pull/26176)
* [[`8fac54a22f`](https://github.com/nodejs/node/commit/8fac54a22f)] - **doc**: fix code lang in repl.md (gengjiawen) [#26075](https://github.com/nodejs/node/pull/26075)
* [[`e5dae20ed6`](https://github.com/nodejs/node/commit/e5dae20ed6)] - **doc**: remove deprecation definition in Collaborator Guide (Rich Trott) [#26157](https://github.com/nodejs/node/pull/26157)
* [[`e108c32865`](https://github.com/nodejs/node/commit/e108c32865)] - **doc**: eliminate use of "note that" from child\_process.md (Rich Trott) [#26141](https://github.com/nodejs/node/pull/26141)
* [[`e506f6a2d6`](https://github.com/nodejs/node/commit/e506f6a2d6)] - **doc**: remove unnecessary italics from child\_process.md (Rich Trott) [#26141](https://github.com/nodejs/node/pull/26141)
* [[`b48a04bc32`](https://github.com/nodejs/node/commit/b48a04bc32)] - **doc**: remove unnecessary bold text from child\_process.md (Rich Trott) [#26141](https://github.com/nodejs/node/pull/26141)
* [[`789b818ad1`](https://github.com/nodejs/node/commit/789b818ad1)] - **doc**: remove unnecessary bold italics from child\_process.md (Rich Trott) [#26141](https://github.com/nodejs/node/pull/26141)
* [[`4d1c87ed6b`](https://github.com/nodejs/node/commit/4d1c87ed6b)] - **doc**: remove all-caps shouting from child\_process.md (Rich Trott) [#26141](https://github.com/nodejs/node/pull/26141)
* [[`c810ced543`](https://github.com/nodejs/node/commit/c810ced543)] - **doc**: wrap child\_process.md at 80 characters (Rich Trott) [#26141](https://github.com/nodejs/node/pull/26141)
* [[`a18b847d18`](https://github.com/nodejs/node/commit/a18b847d18)] - **doc**: improve worker\_threads documentation (Anna Henningsen) [#26110](https://github.com/nodejs/node/pull/26110)
* [[`a9c44372e1`](https://github.com/nodejs/node/commit/a9c44372e1)] - **doc**: consolidate N-API material in Collaborator Guide (Rich Trott) [#26094](https://github.com/nodejs/node/pull/26094)
* [[`82bc68b08e`](https://github.com/nodejs/node/commit/82bc68b08e)] - **doc**: fix notable changes in v11 changelog (Michaël Zasso)
* [[`3971510b66`](https://github.com/nodejs/node/commit/3971510b66)] - **doc**: fix changelog entry (Colin Ihrig) [#26114](https://github.com/nodejs/node/pull/26114)
* [[`2ff1644b34`](https://github.com/nodejs/node/commit/2ff1644b34)] - **doc**: fix notable changes list format for 11.9.0 & 11.10.0 (Kai) [#26129](https://github.com/nodejs/node/pull/26129)
* [[`8814d03d4d`](https://github.com/nodejs/node/commit/8814d03d4d)] - **doc,lib,test**: rename node-report to report (cjihrig) [#26371](https://github.com/nodejs/node/pull/26371)
* [[`0034820f67`](https://github.com/nodejs/node/commit/0034820f67)] - **errors**: add ERR\_INSPECTOR\_COMMAND error (cjihrig) [#26255](https://github.com/nodejs/node/pull/26255)
* [[`030b744941`](https://github.com/nodejs/node/commit/030b744941)] - **esm**: process proxy Symbol.toString fix (Guy Bedford) [#25963](https://github.com/nodejs/node/pull/25963)
* [[`14cf22f860`](https://github.com/nodejs/node/commit/14cf22f860)] - **fs, src, lib**: fix `blksize` & `blocks` on Windows (Richard Lau) [#26056](https://github.com/nodejs/node/pull/26056)
* [[`2595fbc8b1`](https://github.com/nodejs/node/commit/2595fbc8b1)] - **http2**: improve compatibility with http/1 (Sagi Tsofan) [#23908](https://github.com/nodejs/node/pull/23908)
* [[`8a551b9d3b`](https://github.com/nodejs/node/commit/8a551b9d3b)] - **http2**: shrink memory to match read data (Anna Henningsen) [#26201](https://github.com/nodejs/node/pull/26201)
* [[`3bc012373a`](https://github.com/nodejs/node/commit/3bc012373a)] - **inspector**: print all listening addresses (Ben Noordhuis) [#26008](https://github.com/nodejs/node/pull/26008)
* [[`b0c310dcf0`](https://github.com/nodejs/node/commit/b0c310dcf0)] - **inspector**: return Error objects on error (cjihrig) [#26255](https://github.com/nodejs/node/pull/26255)
* [[`be671c3bf5`](https://github.com/nodejs/node/commit/be671c3bf5)] - **inspector**: forward errors from InspectorConsoleCall (Anna Henningsen) [#26113](https://github.com/nodejs/node/pull/26113)
* [[`0c4353a444`](https://github.com/nodejs/node/commit/0c4353a444)] - **inspector**: make sure timer handles are cleaned up (Anna Henningsen) [#26088](https://github.com/nodejs/node/pull/26088)
* [[`bf61050e91`](https://github.com/nodejs/node/commit/bf61050e91)] - **lib**: converted element to lowercase in tty.js (Abhishek Agarwal) [#26121](https://github.com/nodejs/node/pull/26121)
* [[`733beb70ae`](https://github.com/nodejs/node/commit/733beb70ae)] - **lib**: convert legacy process.binding to internalBinding (ZYSzys) [#26095](https://github.com/nodejs/node/pull/26095)
* [[`b25694d7ad`](https://github.com/nodejs/node/commit/b25694d7ad)] - **meta**: update note about building on smartOS 16 (Refael Ackermann) [#25684](https://github.com/nodejs/node/pull/25684)
* [[`6d014a6c3d`](https://github.com/nodejs/node/commit/6d014a6c3d)] - **meta**: remove the useless GitHub Account (MaleDong) [#26146](https://github.com/nodejs/node/pull/26146)
* [[`143b844db2`](https://github.com/nodejs/node/commit/143b844db2)] - **meta**: moving jasnell temporarily to TSC emeritus (jasnell) [#26106](https://github.com/nodejs/node/pull/26106)
* [[`d94f4c23fe`](https://github.com/nodejs/node/commit/d94f4c23fe)] - **module**: fix stat cache (Ruben Bridgewater) [#26266](https://github.com/nodejs/node/pull/26266)
* [[`2a66cd34fa`](https://github.com/nodejs/node/commit/2a66cd34fa)] - **module**: simpler shebang function (Ruben Bridgewater) [#26266](https://github.com/nodejs/node/pull/26266)
* [[`54896a6961`](https://github.com/nodejs/node/commit/54896a6961)] - **module**: revert module.\_compile to original state if module is patched (Ujjwal Sharma) [#21573](https://github.com/nodejs/node/pull/21573)
* [[`b338edbb0a`](https://github.com/nodejs/node/commit/b338edbb0a)] - **module**: use compileFunction over Module.wrap (Ujjwal Sharma) [#21573](https://github.com/nodejs/node/pull/21573)
* [[`e72cb94df6`](https://github.com/nodejs/node/commit/e72cb94df6)] - **(SEMVER-MINOR)** **n-api**: implement date object (Jarrod Connolly) [#25917](https://github.com/nodejs/node/pull/25917)
* [[`2335bcd6e6`](https://github.com/nodejs/node/commit/2335bcd6e6)] - **n-api**: turn NAPI\_CALL\_INTO\_MODULE into a function (Anna Henningsen) [#26128](https://github.com/nodejs/node/pull/26128)
* [[`1ce5e63987`](https://github.com/nodejs/node/commit/1ce5e63987)] - **n-api**: do not call into JS when that is not allowed (Anna Henningsen) [#26127](https://github.com/nodejs/node/pull/26127)
* [[`5b8ac58ed8`](https://github.com/nodejs/node/commit/5b8ac58ed8)] - **path**: refactor code for clarity (Ruben Bridgewater) [#25278](https://github.com/nodejs/node/pull/25278)
* [[`348f1fbcb3`](https://github.com/nodejs/node/commit/348f1fbcb3)] - **path**: refactor for less indentation (Ruben Bridgewater) [#25278](https://github.com/nodejs/node/pull/25278)
* [[`e00c8cd54a`](https://github.com/nodejs/node/commit/e00c8cd54a)] - **path**: simplify code and remove obsolete checks (Ruben Bridgewater) [#25278](https://github.com/nodejs/node/pull/25278)
* [[`55d6b4961a`](https://github.com/nodejs/node/commit/55d6b4961a)] - **path**: refactor logic for to reduce code branches (Ruben Bridgewater) [#25278](https://github.com/nodejs/node/pull/25278)
* [[`6c7cd9ee5a`](https://github.com/nodejs/node/commit/6c7cd9ee5a)] - **path**: minor refactoring (Ruben Bridgewater) [#25278](https://github.com/nodejs/node/pull/25278)
* [[`cccc44b854`](https://github.com/nodejs/node/commit/cccc44b854)] - **path**: refactor more path code for simplicity (Ruben Bridgewater) [#25278](https://github.com/nodejs/node/pull/25278)
* [[`6c44e68f63`](https://github.com/nodejs/node/commit/6c44e68f63)] - **path**: more small refactorings (Ruben Bridgewater) [#25278](https://github.com/nodejs/node/pull/25278)
* [[`b0cde2c4cf`](https://github.com/nodejs/node/commit/b0cde2c4cf)] - **path**: minor refactoring (Ruben Bridgewater) [#25278](https://github.com/nodejs/node/pull/25278)
* [[`d91520724c`](https://github.com/nodejs/node/commit/d91520724c)] - **process**: use common operations to define browser globals (Joyee Cheung) [#26230](https://github.com/nodejs/node/pull/26230)
* [[`b1e739d881`](https://github.com/nodejs/node/commit/b1e739d881)] - **process**: move initialization of node-report into pre\_execution.js (Joyee Cheung) [#26227](https://github.com/nodejs/node/pull/26227)
* [[`57179a0aab`](https://github.com/nodejs/node/commit/57179a0aab)] - **process**: setup signal handler in prepareMainThreadExecution (Joyee Cheung) [#26227](https://github.com/nodejs/node/pull/26227)
* [[`966546ceaa`](https://github.com/nodejs/node/commit/966546ceaa)] - **process**: simplify the setup of async hooks trace events (Joyee Cheung) [#26062](https://github.com/nodejs/node/pull/26062)
* [[`cd10e25bd6`](https://github.com/nodejs/node/commit/cd10e25bd6)] - **process**: move test-process-uptime to parallel (Joyee Cheung) [#26206](https://github.com/nodejs/node/pull/26206)
* [[`fde40116c4`](https://github.com/nodejs/node/commit/fde40116c4)] - **process**: fix calculation in process.uptime() (Joyee Cheung) [#26206](https://github.com/nodejs/node/pull/26206)
* [[`230e98b54a`](https://github.com/nodejs/node/commit/230e98b54a)] - **process**: start coverage collection before bootstrap (Joyee Cheung) [#26006](https://github.com/nodejs/node/pull/26006)
* [[`b5fe27ccc9`](https://github.com/nodejs/node/commit/b5fe27ccc9)] - **process**: delay setup of global exception handlers (Joyee Cheung) [#26061](https://github.com/nodejs/node/pull/26061)
* [[`0d660d9646`](https://github.com/nodejs/node/commit/0d660d9646)] - **readline**: improve Unicode handling (Avi ד) [#25723](https://github.com/nodejs/node/pull/25723)
* [[`4c254d6294`](https://github.com/nodejs/node/commit/4c254d6294)] - **repl**: use object writer for thrown errors (Anna Henningsen) [#26361](https://github.com/nodejs/node/pull/26361)
* [[`2a74a1ed60`](https://github.com/nodejs/node/commit/2a74a1ed60)] - **repl**: hide editor mode if not used in a terminal (Ruben Bridgewater) [#26240](https://github.com/nodejs/node/pull/26240)
* [[`2fa8170e51`](https://github.com/nodejs/node/commit/2fa8170e51)] - **repl**: add new line on ctrl+d (Ruben Bridgewater) [#26240](https://github.com/nodejs/node/pull/26240)
* [[`f636f15315`](https://github.com/nodejs/node/commit/f636f15315)] - **repl**: add more information (Ruben Bridgewater) [#26240](https://github.com/nodejs/node/pull/26240)
* [[`2908e6313b`](https://github.com/nodejs/node/commit/2908e6313b)] - **report**: rename location to trigger (cjihrig) [#26386](https://github.com/nodejs/node/pull/26386)
* [[`0579f4283f`](https://github.com/nodejs/node/commit/0579f4283f)] - **report**: use triggerReport() to handle signals (cjihrig) [#26386](https://github.com/nodejs/node/pull/26386)
* [[`b2c77ec081`](https://github.com/nodejs/node/commit/b2c77ec081)] - **report**: use triggerReport() to handle exceptions (cjihrig) [#26386](https://github.com/nodejs/node/pull/26386)
* [[`b62e2289d9`](https://github.com/nodejs/node/commit/b62e2289d9)] - **report**: add fallback for uv\_getnameinfo() failures (Richard Lau) [#26140](https://github.com/nodejs/node/pull/26140)
* [[`2fe9886f6f`](https://github.com/nodejs/node/commit/2fe9886f6f)] - **report**: fix build warning in node\_report.cc (Richard Lau) [#26265](https://github.com/nodejs/node/pull/26265)
* [[`ba5f31ac45`](https://github.com/nodejs/node/commit/ba5f31ac45)] - **report**: use ru\_stime for system CPU calculation (cjihrig) [#26286](https://github.com/nodejs/node/pull/26286)
* [[`d2d94537b2`](https://github.com/nodejs/node/commit/d2d94537b2)] - **report**: simplify heap space iteration (cjihrig) [#26285](https://github.com/nodejs/node/pull/26285)
* [[`6d2a14d385`](https://github.com/nodejs/node/commit/6d2a14d385)] - **report**: refactor argument validation (cjihrig) [#26276](https://github.com/nodejs/node/pull/26276)
* [[`8e2cc5e440`](https://github.com/nodejs/node/commit/8e2cc5e440)] - **report**: refactor triggerReport() (cjihrig) [#26268](https://github.com/nodejs/node/pull/26268)
* [[`8a40468635`](https://github.com/nodejs/node/commit/8a40468635)] - **report**: remove verbose setting (cjihrig) [#26195](https://github.com/nodejs/node/pull/26195)
* [[`0e89d7add6`](https://github.com/nodejs/node/commit/0e89d7add6)] - **report**: simplify OnFatalError() handling (cjihrig) [#26191](https://github.com/nodejs/node/pull/26191)
* [[`633c1eac29`](https://github.com/nodejs/node/commit/633c1eac29)] - **report**: simplify TriggerNodeReport() (cjihrig) [#26174](https://github.com/nodejs/node/pull/26174)
* [[`fc9ba36fb2`](https://github.com/nodejs/node/commit/fc9ba36fb2)] - **src**: fix typo in callback.cc (gengjiawen) [#26337](https://github.com/nodejs/node/pull/26337)
* [[`63942de82c`](https://github.com/nodejs/node/commit/63942de82c)] - **src**: extra-semi warning in node\_platform.h (Jeremy Apthorp) [#26330](https://github.com/nodejs/node/pull/26330)
* [[`cb62c24e1b`](https://github.com/nodejs/node/commit/cb62c24e1b)] - **src**: reduce to simple `const char\*` in OptionsParser (ZYSzys) [#26297](https://github.com/nodejs/node/pull/26297)
* [[`3093617c0e`](https://github.com/nodejs/node/commit/3093617c0e)] - **src**: remove unused variable (cjihrig) [#26386](https://github.com/nodejs/node/pull/26386)
* [[`b216f44513`](https://github.com/nodejs/node/commit/b216f44513)] - **src**: remove unnecessary function declaration (cjihrig) [#26386](https://github.com/nodejs/node/pull/26386)
* [[`cb2cbf2eca`](https://github.com/nodejs/node/commit/cb2cbf2eca)] - **src**: remove already elevated Isolate namespce (Juan José Arboleda) [#26294](https://github.com/nodejs/node/pull/26294)
* [[`2438a4350d`](https://github.com/nodejs/node/commit/2438a4350d)] - **src**: remove unused macro in env.cc (gengjiawen) [#26273](https://github.com/nodejs/node/pull/26273)
* [[`4df82f0f1b`](https://github.com/nodejs/node/commit/4df82f0f1b)] - **src**: remove unused macro in node\_http2.h (gengjiawen) [#26204](https://github.com/nodejs/node/pull/26204)
* [[`af2a6935ab`](https://github.com/nodejs/node/commit/af2a6935ab)] - **src**: remove redundant cast in PipeWrap::Fchmod (gengjiawen) [#26242](https://github.com/nodejs/node/pull/26242)
* [[`06d592c551`](https://github.com/nodejs/node/commit/06d592c551)] - **src**: simplify native immediate by using v8::Global (Anna Henningsen) [#26254](https://github.com/nodejs/node/pull/26254)
* [[`9b4eec0aad`](https://github.com/nodejs/node/commit/9b4eec0aad)] - **src**: allow not materializing ArrayBuffers from C++ (Anna Henningsen) [#26301](https://github.com/nodejs/node/pull/26301)
* [[`30f0a3b4bd`](https://github.com/nodejs/node/commit/30f0a3b4bd)] - **src**: remove dead inspector code (Anna Henningsen) [#26295](https://github.com/nodejs/node/pull/26295)
* [[`c37b6796df`](https://github.com/nodejs/node/commit/c37b6796df)] - **src**: remove unused Converter object (Anna Henningsen) [#26243](https://github.com/nodejs/node/pull/26243)
* [[`6f9ab5e15b`](https://github.com/nodejs/node/commit/6f9ab5e15b)] - **src**: remove redundant cast in method AfterStringPath (gengjiawen) [#26218](https://github.com/nodejs/node/pull/26218)
* [[`33d6a3fcb7`](https://github.com/nodejs/node/commit/33d6a3fcb7)] - **src**: clean up `StreamPipe` in destructor (Anna Henningsen) [#26256](https://github.com/nodejs/node/pull/26256)
* [[`75ae77d99f`](https://github.com/nodejs/node/commit/75ae77d99f)] - **src**: do not access Environment-owned handles after cleanup (Anna Henningsen) [#26256](https://github.com/nodejs/node/pull/26256)
* [[`d6759db15b`](https://github.com/nodejs/node/commit/d6759db15b)] - **src**: remove cast for unsupported openssl (Sam Roberts) [#26305](https://github.com/nodejs/node/pull/26305)
* [[`1abe1d1c06`](https://github.com/nodejs/node/commit/1abe1d1c06)] - **src**: track memory retainer fields (Gireesh Punathil) [#26161](https://github.com/nodejs/node/pull/26161)
* [[`3e0978d7a3`](https://github.com/nodejs/node/commit/3e0978d7a3)] - **src**: clean unused macro in inspector\_socket.cc (gengjiawen) [#26158](https://github.com/nodejs/node/pull/26158)
* [[`4001b24f79`](https://github.com/nodejs/node/commit/4001b24f79)] - **src**: remove unimplemented method in class SSLWrap (gengjiawen) [#26203](https://github.com/nodejs/node/pull/26203)
* [[`8b515b24af`](https://github.com/nodejs/node/commit/8b515b24af)] - **src**: apply clang-tidy rule modernize-deprecated-headers (gengjiawen) [#26159](https://github.com/nodejs/node/pull/26159)
* [[`3c11b4eec2`](https://github.com/nodejs/node/commit/3c11b4eec2)] - **src**: allocate Buffer memory using ArrayBuffer allocator (Anna Henningsen) [#26207](https://github.com/nodejs/node/pull/26207)
* [[`282607644b`](https://github.com/nodejs/node/commit/282607644b)] - **src**: add allocation utils to env (Anna Henningsen) [#26207](https://github.com/nodejs/node/pull/26207)
* [[`238fa5704b`](https://github.com/nodejs/node/commit/238fa5704b)] - **src**: add debugging array allocator (Anna Henningsen) [#26207](https://github.com/nodejs/node/pull/26207)
* [[`437bb25d92`](https://github.com/nodejs/node/commit/437bb25d92)] - **src**: make IsolateData store ArrayBufferAllocator (Anna Henningsen) [#26207](https://github.com/nodejs/node/pull/26207)
* [[`68accb5b04`](https://github.com/nodejs/node/commit/68accb5b04)] - **src**: use smart pointer in UDPWrap::OnSend (Daniel Bevenius) [#26233](https://github.com/nodejs/node/pull/26233)
* [[`3abdcfc813`](https://github.com/nodejs/node/commit/3abdcfc813)] - **src**: remove unimplemented method in class StreamPipe (gengjiawen) [#26202](https://github.com/nodejs/node/pull/26202)
* [[`7e26ca6750`](https://github.com/nodejs/node/commit/7e26ca6750)] - **src**: simplify AliasedBuffer lifetime management (Anna Henningsen) [#26196](https://github.com/nodejs/node/pull/26196)
* [[`831aa9acb6`](https://github.com/nodejs/node/commit/831aa9acb6)] - **src**: make `node::SignalWrap::OnSignal` into lambda (Gireesh Punathil) [#26184](https://github.com/nodejs/node/pull/26184)
* [[`619b5e7c2e`](https://github.com/nodejs/node/commit/619b5e7c2e)] - **src**: simplify loop arithmetic in `GetCPUInfo` (Gireesh Punathil) [#26183](https://github.com/nodejs/node/pull/26183)
* [[`ddd71f4a92`](https://github.com/nodejs/node/commit/ddd71f4a92)] - **src**: move function from header to source file (Ben Noordhuis) [#26173](https://github.com/nodejs/node/pull/26173)
* [[`5cc2574fac`](https://github.com/nodejs/node/commit/5cc2574fac)] - **src**: move async hooks trace events setup to pre\_execution.js (Joyee Cheung) [#26062](https://github.com/nodejs/node/pull/26062)
* [[`8881c0baaa`](https://github.com/nodejs/node/commit/8881c0baaa)] - **src**: simplify InspectorConsoleCall (Anna Henningsen) [#26168](https://github.com/nodejs/node/pull/26168)
* [[`c6d5af53be`](https://github.com/nodejs/node/commit/c6d5af53be)] - **src**: move req\_wrap\_queue to base class of ReqWrap (Anna Henningsen) [#26148](https://github.com/nodejs/node/pull/26148)
* [[`a39cd45ce8`](https://github.com/nodejs/node/commit/a39cd45ce8)] - **src**: remove `process.binding('config').fipsForced` (Joyee Cheung) [#26178](https://github.com/nodejs/node/pull/26178)
* [[`bd40a127f9`](https://github.com/nodejs/node/commit/bd40a127f9)] - **src**: only call .ReThrow() if not terminating (Anna Henningsen) [#26130](https://github.com/nodejs/node/pull/26130)
* [[`6b7d8369e3`](https://github.com/nodejs/node/commit/6b7d8369e3)] - **src**: add missing includes for vtune build (Uttam Pawar) [#26136](https://github.com/nodejs/node/pull/26136)
* [[`25ddbc9a36`](https://github.com/nodejs/node/commit/25ddbc9a36)] - **src**: apply clang-tidy rule performance-unnecessary-value-param (gengjiawen) [#26042](https://github.com/nodejs/node/pull/26042)
* [[`82df851bb5`](https://github.com/nodejs/node/commit/82df851bb5)] - **src**: unify uptime base used across the code base (Joyee Cheung) [#26016](https://github.com/nodejs/node/pull/26016)
* [[`778db675c1`](https://github.com/nodejs/node/commit/778db675c1)] - **src**: remove invalid casts in options parser (Anna Henningsen) [#26139](https://github.com/nodejs/node/pull/26139)
* [[`4ca07898d7`](https://github.com/nodejs/node/commit/4ca07898d7)] - **src**: use PauseOnNextJavascriptStatement to implement --inspect-brk-node (Joyee Cheung) [#26034](https://github.com/nodejs/node/pull/26034)
* [[`e6949b4241`](https://github.com/nodejs/node/commit/e6949b4241)] - **src**: apply clang-tidy rule modernize-use-override (gengjiawen) [#26103](https://github.com/nodejs/node/pull/26103)
* [[`d550de4fe1`](https://github.com/nodejs/node/commit/d550de4fe1)] - **src**: remove inspector main\_thread\_request\_ field (Anna Henningsen) [#26137](https://github.com/nodejs/node/pull/26137)
* [[`ee71952a25`](https://github.com/nodejs/node/commit/ee71952a25)] - **src**: check HasCaught() in JSStream calls (Anna Henningsen) [#26124](https://github.com/nodejs/node/pull/26124)
* [[`f44f33569d`](https://github.com/nodejs/node/commit/f44f33569d)] - **src**: extract common sockaddr creation code (Daniel Bevenius) [#26070](https://github.com/nodejs/node/pull/26070)
* [[`cbd3cf083a`](https://github.com/nodejs/node/commit/cbd3cf083a)] - **src**: add debug CHECKs against empty handles (Anna Henningsen) [#26125](https://github.com/nodejs/node/pull/26125)
* [[`0408966a9d`](https://github.com/nodejs/node/commit/0408966a9d)] - **src**: remove unused macro in node\_file.cc (gengjiawen) [#26073](https://github.com/nodejs/node/pull/26073)
* [[`497d9d8ab2`](https://github.com/nodejs/node/commit/497d9d8ab2)] - **src**: use same parameter name in node\_report.cc (gengjiawen) [#26046](https://github.com/nodejs/node/pull/26046)
* [[`e314681420`](https://github.com/nodejs/node/commit/e314681420)] - **src**: use more stable cast where possible (Gireesh Punathil) [#26052](https://github.com/nodejs/node/pull/26052)
* [[`7612574e42`](https://github.com/nodejs/node/commit/7612574e42)] - **stream**: make \_read() be called indefinitely if the user wants so (Matteo Collina) [#26135](https://github.com/nodejs/node/pull/26135)
* [[`50e42c9d64`](https://github.com/nodejs/node/commit/50e42c9d64)] - **test**: improve test coverage in perf\_hooks (Juan José Arboleda) [#26290](https://github.com/nodejs/node/pull/26290)
* [[`a41138b0cf`](https://github.com/nodejs/node/commit/a41138b0cf)] - **test**: remove duplicated buffer negative allocation test (ZYSzys) [#26160](https://github.com/nodejs/node/pull/26160)
* [[`93d7fa3df3`](https://github.com/nodejs/node/commit/93d7fa3df3)] - **test**: only inspect on failure (Ruben Bridgewater) [#26360](https://github.com/nodejs/node/pull/26360)
* [[`91b61452c3`](https://github.com/nodejs/node/commit/91b61452c3)] - **test**: always activate colors if necessary (Ruben Bridgewater) [#26264](https://github.com/nodejs/node/pull/26264)
* [[`11bd5e07cb`](https://github.com/nodejs/node/commit/11bd5e07cb)] - **test**: rename node-report suite to report (cjihrig) [#26371](https://github.com/nodejs/node/pull/26371)
* [[`7ccffcbcb6`](https://github.com/nodejs/node/commit/7ccffcbcb6)] - **test**: improve validation of report output (cjihrig) [#26289](https://github.com/nodejs/node/pull/26289)
* [[`4561cf351f`](https://github.com/nodejs/node/commit/4561cf351f)] - **test**: verify heap buffer allocations occur (Anna Henningsen) [#26301](https://github.com/nodejs/node/pull/26301)
* [[`0c8e9ee62e`](https://github.com/nodejs/node/commit/0c8e9ee62e)] - **test**: fix for activities in tick objects prune function (Alexander Sattelmaier) [#26163](https://github.com/nodejs/node/pull/26163)
* [[`69154e405c`](https://github.com/nodejs/node/commit/69154e405c)] - **test**: refactor tick objects prune function (Alexander Sattelmaier) [#26163](https://github.com/nodejs/node/pull/26163)
* [[`d8f5f55b78`](https://github.com/nodejs/node/commit/d8f5f55b78)] - **test**: eliminate port collision (Gireesh Punathil) [#26298](https://github.com/nodejs/node/pull/26298)
* [[`88256d7ba2`](https://github.com/nodejs/node/commit/88256d7ba2)] - **test**: simplify node-report/test-exception.js (cjihrig) [#26277](https://github.com/nodejs/node/pull/26277)
* [[`e8995d1b80`](https://github.com/nodejs/node/commit/e8995d1b80)] - **test**: increase getReport() coverage (cjihrig) [#26276](https://github.com/nodejs/node/pull/26276)
* [[`33fe892ec6`](https://github.com/nodejs/node/commit/33fe892ec6)] - **test**: increase triggerReport() coverage (cjihrig) [#26268](https://github.com/nodejs/node/pull/26268)
* [[`a382b52fd8`](https://github.com/nodejs/node/commit/a382b52fd8)] - **test**: consolidate triggerReport() tests (cjihrig) [#26268](https://github.com/nodejs/node/pull/26268)
* [[`6f9a764b52`](https://github.com/nodejs/node/commit/6f9a764b52)] - **test**: remove node-report/test-api.js (cjihrig) [#26219](https://github.com/nodejs/node/pull/26219)
* [[`bc114152d0`](https://github.com/nodejs/node/commit/bc114152d0)] - **test**: simplify test-api-nohooks.js (cjihrig) [#26217](https://github.com/nodejs/node/pull/26217)
* [[`ca18525896`](https://github.com/nodejs/node/commit/ca18525896)] - **test**: improve performance of test-crypto-timing-safe-equal-benchmarks (Rich Trott) [#26237](https://github.com/nodejs/node/pull/26237)
* [[`28758b8d69`](https://github.com/nodejs/node/commit/28758b8d69)] - **test**: add test for dynamically enabling node.async\_hooks tracing (Joyee Cheung) [#26062](https://github.com/nodejs/node/pull/26062)
* [[`dcbd907142`](https://github.com/nodejs/node/commit/dcbd907142)] - **test**: add test for node.async\_hooks tracing in workers (Joyee Cheung) [#26062](https://github.com/nodejs/node/pull/26062)
* [[`007b2fa198`](https://github.com/nodejs/node/commit/007b2fa198)] - **test**: increase run time in test-worker-prof (Anna Henningsen) [#26172](https://github.com/nodejs/node/pull/26172)
* [[`a1fcde035e`](https://github.com/nodejs/node/commit/a1fcde035e)] - **test**: simplify test-api-getreport.js (cjihrig) [#26169](https://github.com/nodejs/node/pull/26169)
* [[`818b280a39`](https://github.com/nodejs/node/commit/818b280a39)] - **test**: remove unnecessary default tmpdir value in test (Rich Trott) [#26177](https://github.com/nodejs/node/pull/26177)
* [[`59ca9e9ccf`](https://github.com/nodejs/node/commit/59ca9e9ccf)] - **test**: consolidate assertions in ipv6only test (Rich Trott) [#26149](https://github.com/nodejs/node/pull/26149)
* [[`38a87d5521`](https://github.com/nodejs/node/commit/38a87d5521)] - **test**: increase coverage of node\_report\_module.cc (Richard Lau) [#26116](https://github.com/nodejs/node/pull/26116)
* [[`76c2f4f46b`](https://github.com/nodejs/node/commit/76c2f4f46b)] - **test**: simplify test-worker-syntax-error (Rich Trott) [#26144](https://github.com/nodejs/node/pull/26144)
* [[`441b5453a0`](https://github.com/nodejs/node/commit/441b5453a0)] - **test**: fix flaky test-worker-ref-onexit (Anna Henningsen) [#26170](https://github.com/nodejs/node/pull/26170)
* [[`d3525d7505`](https://github.com/nodejs/node/commit/d3525d7505)] - **test**: add --test-root option to test.py (Yang Guo) [#26093](https://github.com/nodejs/node/pull/26093)
* [[`a920721175`](https://github.com/nodejs/node/commit/a920721175)] - **test**: silence compiler warning in openssl-binding (Daniel Bevenius) [#26067](https://github.com/nodejs/node/pull/26067)
* [[`2d0242a69b`](https://github.com/nodejs/node/commit/2d0242a69b)] - **test**: increase coverage for assertion\_error.js (Rich Trott) [#26065](https://github.com/nodejs/node/pull/26065)
* [[`dd60cd60b3`](https://github.com/nodejs/node/commit/dd60cd60b3)] - **test**: add arg to narrow http benchmark test (Refael Ackermann) [#26101](https://github.com/nodejs/node/pull/26101)
* [[`fbf6dd558a`](https://github.com/nodejs/node/commit/fbf6dd558a)] - **test,inspector**: add heap allocation tracker test (Anna Henningsen) [#26089](https://github.com/nodejs/node/pull/26089)
* [[`db94ab778f`](https://github.com/nodejs/node/commit/db94ab778f)] - **test,worker**: posting undefined/null message to message port (legendecas) [#26123](https://github.com/nodejs/node/pull/26123)
* [[`d1e3724b5d`](https://github.com/nodejs/node/commit/d1e3724b5d)] - **test,worker**: add more tests for worker.ref()/.unref() (Anna Henningsen) [#26083](https://github.com/nodejs/node/pull/26083)
* [[`96a5765491`](https://github.com/nodejs/node/commit/96a5765491)] - **tools**: update extend to 3.0.2 (Rich Trott) [#26392](https://github.com/nodejs/node/pull/26392)
* [[`6e9a7e1048`](https://github.com/nodejs/node/commit/6e9a7e1048)] - **tools**: remove unneeded .gitignore entries (Rich Trott) [#26370](https://github.com/nodejs/node/pull/26370)
* [[`123fad6e1c`](https://github.com/nodejs/node/commit/123fad6e1c)] - **tools**: update babel-eslint to 10.0.1 (Rich Trott) [#26347](https://github.com/nodejs/node/pull/26347)
* [[`347dd99251`](https://github.com/nodejs/node/commit/347dd99251)] - **tools**: update eslint-plugin-markdown to 1.0.0 (Rich Trott) [#26345](https://github.com/nodejs/node/pull/26345)
* [[`adcbcf5bd6`](https://github.com/nodejs/node/commit/adcbcf5bd6)] - **tools**: use latest rather than next for markdown linting plugin (Rich Trott) [#26345](https://github.com/nodejs/node/pull/26345)
* [[`0080350f1a`](https://github.com/nodejs/node/commit/0080350f1a)] - **tools**: update markdown linter (Rich Trott) [#26281](https://github.com/nodejs/node/pull/26281)
* [[`dff0149d57`](https://github.com/nodejs/node/commit/dff0149d57)] - **tools**: update ESLint to 5.14.1 (cjihrig) [#26190](https://github.com/nodejs/node/pull/26190)
* [[`28d607444d`](https://github.com/nodejs/node/commit/28d607444d)] - **tools**: update ESLint to 5.14.0 (cjihrig) [#26142](https://github.com/nodejs/node/pull/26142)
* [[`1766b8c341`](https://github.com/nodejs/node/commit/1766b8c341)] - **trace_events**: fix trace events JS API writing (Kelvin Jin) [#24945](https://github.com/nodejs/node/pull/24945)
* [[`34c685b406`](https://github.com/nodejs/node/commit/34c685b406)] - **tracing**: use ‘real’ atomics (Anna Henningsen) [#26156](https://github.com/nodejs/node/pull/26156)
* [[`b6355ef602`](https://github.com/nodejs/node/commit/b6355ef602)] - **tty**: improve color detection (Ruben Bridgewater) [#26264](https://github.com/nodejs/node/pull/26264)
* [[`001785520a`](https://github.com/nodejs/node/commit/001785520a)] - **url**: handle quasi-WHATWG URLs in urlToOptions() (cjihrig) [#26226](https://github.com/nodejs/node/pull/26226)
* [[`6828fbb2ef`](https://github.com/nodejs/node/commit/6828fbb2ef)] - **(SEMVER-MINOR)** **util**: group array elements together (Ruben Bridgewater) [#26269](https://github.com/nodejs/node/pull/26269)
* [[`4500ed85e9`](https://github.com/nodejs/node/commit/4500ed85e9)] - **(SEMVER-MINOR)** **util**: add compact depth mode (Ruben Bridgewater) [#26269](https://github.com/nodejs/node/pull/26269)
* [[`34905fc2b9`](https://github.com/nodejs/node/commit/34905fc2b9)] - **util**: mark iterator entries as such (Ruben Bridgewater) [#26222](https://github.com/nodejs/node/pull/26222)
* [[`4bf58ac13d`](https://github.com/nodejs/node/commit/4bf58ac13d)] - **util**: update set iterator entries inspection (Ruben Bridgewater) [#25941](https://github.com/nodejs/node/pull/25941)
* [[`7d66d47dba`](https://github.com/nodejs/node/commit/7d66d47dba)] - **vm**: do not overwrite error when creating context (Anna Henningsen) [#26112](https://github.com/nodejs/node/pull/26112)
* [[`8cf4170c94`](https://github.com/nodejs/node/commit/8cf4170c94)] - **worker**: provide process.execArgv (Anna Henningsen) [#26267](https://github.com/nodejs/node/pull/26267)
* [[`6fdc502a32`](https://github.com/nodejs/node/commit/6fdc502a32)] - **worker**: make MessagePort `uv\_async\_t` inline field (Anna Henningsen) [#26271](https://github.com/nodejs/node/pull/26271)
* [[`51f01aa25b`](https://github.com/nodejs/node/commit/51f01aa25b)] - **worker**: remove MessagePort::AddToIncomingQueue (Anna Henningsen) [#26271](https://github.com/nodejs/node/pull/26271)
* [[`74d11e7d0e`](https://github.com/nodejs/node/commit/74d11e7d0e)] - **worker**: refactor thread life cycle management (Gireesh Punathil) [#26099](https://github.com/nodejs/node/pull/26099)
* [[`20dc172011`](https://github.com/nodejs/node/commit/20dc172011)] - **worker**: copy transferList ArrayBuffers on unknown allocator (Anna Henningsen) [#26207](https://github.com/nodejs/node/pull/26207)
* [[`7e7023373a`](https://github.com/nodejs/node/commit/7e7023373a)] - **worker**: serialize errors if stack getter throws (Rich Trott) [#26145](https://github.com/nodejs/node/pull/26145)
* [[`a9a2c5869c`](https://github.com/nodejs/node/commit/a9a2c5869c)] - **(SEMVER-MINOR)** **worker**: improve integration with native addons (Anna Henningsen) [#26175](https://github.com/nodejs/node/pull/26175)
* [[`dab3d71243`](https://github.com/nodejs/node/commit/dab3d71243)] - **worker**: ignore --abort-on-uncaught-exception for terminate() (Anna Henningsen) [#26111](https://github.com/nodejs/node/pull/26111)
* [[`dab64bb0e8`](https://github.com/nodejs/node/commit/dab64bb0e8)] - **worker**: spin uv\_run twice before closing loop (Anna Henningsen) [#26138](https://github.com/nodejs/node/pull/26138)
* [[`24debc9d5c`](https://github.com/nodejs/node/commit/24debc9d5c)] - **worker**: do not add removed methods to MessagePort (Anna Henningsen) [#26109](https://github.com/nodejs/node/pull/26109)
* [[`8045e40917`](https://github.com/nodejs/node/commit/8045e40917)] - **worker**: remove duplicate call (Gireesh Punathil) [#26104](https://github.com/nodejs/node/pull/26104)
* [[`69298713af`](https://github.com/nodejs/node/commit/69298713af)] - **worker**: switch to internal assert module (Rich Trott) [#26091](https://github.com/nodejs/node/pull/26091)
* [[`77a944cdee`](https://github.com/nodejs/node/commit/77a944cdee)] - **worker**: use fake MessageEvent for port.onmessage (Anna Henningsen) [#26082](https://github.com/nodejs/node/pull/26082)
* [[`851a691678`](https://github.com/nodejs/node/commit/851a691678)] - **zlib**: report premature ends earlier (Anna Henningsen) [#26363](https://github.com/nodejs/node/pull/26363)

BridgeAR commented:
@nodejs/releasers it was originally planned to release a new current version today. However, the CI is still on locked due to the upcoming security release. Should someone else trigger the CI and release this before the security release or shall we wait until that's out of the way?

richardlau commented:
I'd suggest delaying to avoid potential confusion. v10.15.2 was also due today but [has been delayed](https://github.com/nodejs/node/pull/26063#issuecomment-467300367).

BridgeAR commented:
I just updated the proposal. @nodejs/releasers PTAL

CI https://ci.nodejs.org/job/node-test-pull-request/21202/
V8-CI https://ci.nodejs.org/job/node-test-commit-v8-linux/2132/
CITGM https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker/1758/

targos commented:
- Commit message shouldn't include you GH handle.
- There are REPLACEME placeholders to replace in `doc/api/n-api.md` and `doc/api/util.md`.
- Please add the notable changes to the commit message

BridgeAR commented:
@targos done

richardlau commented:
withoutintl and withoutssl builds are failing:

https://ci.nodejs.org/job/node-test-commit-linux-containered/11042/nodes=ubuntu1604_sharedlibs_withoutintl_x64/console
```
19:36:59 /home/iojs/build/workspace/node-test-commit-linux-containered/out/Release/obj.target/node_lib/src/node_config.o: In function `node::Initialize(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Context>)':
19:36:59 node_config.cc:(.text+0x201): undefined reference to `node::EnvironmentOptions::debug_options() const'
19:36:59 node_config.cc:(.text+0x356): undefined reference to `node::EnvironmentOptions::debug_options() const'
19:36:59 node_config.cc:(.text+0x3c1): undefined reference to `node::EnvironmentOptions::debug_options() const'
19:36:59 collect2: error: ld returned 1 exit status
```

https://ci.nodejs.org/job/node-test-commit-linux-containered/11042/nodes=ubuntu1604_sharedlibs_withoutssl_x64/console
```
19:11:14 /home/iojs/build/workspace/node-test-commit-linux-containered/out/Release/obj.target/node_lib/src/node_config.o: In function `node::Initialize(v8::Local<v8::Object>, v8::Local<v8::Value>, v8::Local<v8::Context>)':
19:11:14 node_config.cc:(.text+0x377): undefined reference to `node::EnvironmentOptions::debug_options() const'
19:11:14 node_config.cc:(.text+0x4c6): undefined reference to `node::EnvironmentOptions::debug_options() const'
19:11:14 node_config.cc:(.text+0x531): undefined reference to `node::EnvironmentOptions::debug_options() const'
19:11:14 collect2: error: ld returned 1 exit status
```

BridgeAR commented:
@richardlau can you pinpoint the commit that causes the issue?

richardlau commented:
> 
> 
> @richardlau can you pinpoint the commit that causes the issue?

Currently bisecting.

BridgeAR commented:
@targos suspects https://github.com/nodejs/node/pull/25999 to be the code that fails here. It could be a commit that relies on this code being removed.

However, we fail to find the cause of it.

BridgeAR commented:
It does compile for me when I include the #25999. @joyeecheung @addaleax PTAL

richardlau commented:
> 
> 
> > @richardlau can you pinpoint the commit that causes the issue?
> 
> Currently bisecting.

With this script and `git bisect run build.sh`:
```bash
#!/usr/bin/env bash

./configure --without-intl --without-ssl && make clean -j 128 && make -j 128
```

Points to:
```console
668ca360a00c5435b0a40e09fde57228808d8be2 is the first bad commit
commit 668ca360a00c5435b0a40e09fde57228808d8be2
Author: Refael Ackermann <refack@gmail.com>
Date:   Mon Feb 25 15:42:28 2019 -0500

    src: fix warnings around node_options

    * header explicit usage, order, and reduce use of `*-inl.h`
    * pointer -> const reference when possible
    * no variable recyclicng
    * `std::begin/end` prefered over `instance.begin/end`
    * `USE` for explicit unused resaults

    PR-URL: https://github.com/nodejs/node/pull/26280
    Fixes: https://github.com/nodejs/node/issues/25593
    Reviewed-By: Joyee Cheung <joyeec9h3@gmail.com>

:040000 040000 4905b4668f63e2594c05d9c6e2ce2f39a27b993d ff0bb87cdcd459cbd9f16e28da4c129b39d54086 M  src
bisect run success
-bash-4.2$
```

addaleax commented:
That makes sense, the un-inclusion of `node_options-inl.h` causes this. There are probably compiler warnings popping up during compilation as well.

You can exclude that commit from the release proposal, I think that’s the easiest short-term solution.

addaleax commented:
For the notable changes: I’d maybe mention in the `util` part that those two refer to `util.inspect()`?

I would maybe also include 77a944cdee since it’s an API change (not to an method of an experimental API that isn’t documented our side, but it is documented on the Web side). Something like:
> `MessagePort.prototype.onmessage` takes arguments closer to the Web specification now.

And generally: Thanks for preparing this :slightly_smiling_face:

BridgeAR commented:
I backed out the mentioned PR that caused the build failures. It won't be possible to push the release today anymore, so I also updated the release date.

@addaleax I updated the notable changes as suggested. PTAL.

CI https://ci.nodejs.org/job/node-test-pull-request/21233/
V8-CI https://ci.nodejs.org/job/node-test-commit-v8-linux/2138/

Earlier I went through the CITGM results and could not find anything related to this release. There are a couple of failures that pop up frequently at the moment and I opened some PRs to fix them.

addaleax commented:
@bridgear Fwiw, the first commit in that PR seemed to be okay. The notable changes LGTM :+1:

BridgeAR commented:
@addaleax AFAIK we always include / exclude whole PRs and not individual commits from a PR.

BridgeAR commented:
Resumed CI https://ci.nodejs.org/job/node-test-commit/26425/
Resumed V8-CI https://ci.nodejs.org/job/node-test-commit-v8-linux/2141/

@nodejs/benchmarking-admins the CI failed at least twice on the job e.g. https://ci.nodejs.org/job/node-test-commit-v8-linux/2138/nodes=benchmark,v8test=v8test/console

BridgeAR commented:
@nodejs/releasers @nodejs/build any opinion on the benchmark failures from the V8-CI? Should this block the release? It seems reproducible on each run.

targos commented:
Run against `master` to compare: https://ci.nodejs.org/job/node-test-commit-v8-linux/2142/

BridgeAR commented:
@targos it does

targos commented:
So I guess it's an infra issue and we can go ahead and ignore it?

BridgeAR commented:
Once more resuming the CI https://ci.nodejs.org/job/node-test-commit/26434/

BridgeAR commented:
Seems like there is one legitimate test failure for the `--without-ssl` builds.

I am looking into it right now.

BridgeAR commented:
It seems a deprecation warning is emitted twice for such builds while requireing `'node-inspect/lib/internal/inspect_client'`. https://ci.nodejs.org/job/node-test-commit-linux-containered/11111/nodes=ubuntu1604_sharedlibs_withoutssl_x64/console

I would like to ignore that test for now as it does not seem critical to block the release. Any other opinions? @nodejs/releasers @nodejs/tsc

BridgeAR commented:
I am not able to reproduce the issue locally by the way.

bcoe commented:
@BridgeAR I'd like to advocate that we backport https://github.com/nodejs/node/commit/c78788a0219cc950be7159e65b9e8db01e9c1a7a, this makes coverage work significantly better.

BridgeAR commented:
Rebuild linux-containered https://ci.nodejs.org/job/node-test-commit-linux-containered/11113/

BridgeAR commented:
@bcoe we plan on releasing a new version in a week again and I would rather get this release out as it is as long as nothing is blocking it.

bcoe commented:
sounds good @BridgeAR 👍 just want to make sure some of these backports are on our radar. I'm going to also see if we could port any of these patches back to `8.x`.

richardlau commented:
> 
> 
> It seems a deprecation warning is emitted twice for such builds while requireing `'node-inspect/lib/internal/inspect_client'`. https://ci.nodejs.org/job/node-test-commit-linux-containered/11111/nodes=ubuntu1604_sharedlibs_withoutssl_x64/console
> 
> I would like to ignore that test for now as it does not seem critical to block the release. Any other opinions? @nodejs/releasers @nodejs/tsc

We have only just enabled the `--without-ssl` builds in the last week: https://github.com/nodejs/build/issues/1574

The test in question doesn't exist in `master` (removed in https://github.com/nodejs/node/pull/25138 which is semver-major). I kicked off https://ci.nodejs.org/job/node-test-commit-linux-containered/11117/ against `v11.x` (i.e. 11.10.1) and the test failed there so it's not a new regression in this proposed release so I'm okay with ignoring this for now but we do need to address it so the builds pass for the next 11.x release.

refack commented:
a bit off-topic, but, there's lint in the `v11.x-staging` branch - https://travis-ci.com/nodejs/node/builds/104258827

BridgeAR commented:
@refack this has nothing to do with that release. I am working currently working on the staging branch to have a working branch to open a new release proposal but there were lots and lots of conflicts this time and I had to bisect a few times to find the commit that caused issues. So I am still going to force-push all issues out.

ZauberNerd commented:
@BridgeAR with V11.11.0 something changed in regards to how `require` handles failed attempts:
With v11.10.1 and below the following worked:
```javascript
try {
  require('./non-existant');
} catch (e) {}
require('fs').writeFileSync('./non-existant', '{}');
require('./non-existant');
```
The second `require` attempt does not throw but instead resolves to the file.
With Node v11.11.0 the second (and all following) `require` calls throw an error that the module has not been found. I tried looking into deleting the `require.cache` but it seems like the cache does not have entries for failed require attempts.

I have found the following three commits that touch the module system, which might be causing this behaviour:
- https://github.com/nodejs/node/commit/d94f4c23fe
- https://github.com/nodejs/node/commit/54896a6961
- https://github.com/nodejs/node/commit/b338edbb0a

I'm currently trying to compile each of these commits to find out which one introduced this behaviour and will open a proper issue once I have found out what actually happens here.

BridgeAR commented:
@ZauberNerd thanks for reporting that. It should indeed resolve and I am pretty sure it's d94f4c2 that broke this.

Please open a separate issue for this though. Thanks!
